src/vehicle.h

```diff
@@ -701,31 +704,54 @@ class vehicle : public JsonSerializer, public JsonDeserializer
     // Loop through engines and generate noise and smoke for each one
     void noise_and_smoke( double load, double time = 6.0 );
 
-    // Calculate area covered by wheels and, optionally count number of wheels
-    float wheels_area (int *cnt = 0) const;
+    /**
+     * Calculates the sum of the area under the wheels of the vehicle.
+     * @param boat If true, calculates the area under "wheels" that allow swimming.
+     */
+    float wheel_area( bool boat ) const;
 
-    // Combined coefficient of aerodynamic and wheel friction resistance of vehicle, 0-1.0.
-    // 1.0 means it's ideal form and have no resistance at all. 0 -- it won't move
-    float k_dynamics () const;
+    /**
+     * Combined coefficient of aerodynamic and wheel friction resistance of vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the dynamics coefficient.
+     */
+    float k_dynamics() const;
 
-    // Components of the dynamic coefficient
+    /**
+     * Wheel friction coefficient of the vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the friction coefficient.
+     */
     float k_friction () const;
+
+    /**
+     * Air friction coefficient of the vehicle.
+     * @returns A value from 0.0 to 1.0 (exclusive) describing the aerodynamics coefficient.
+     */
     float k_aerodynamics () const;
 
-    // Coefficient of mass, 0-1.0.
-    // 1.0 means mass won't slow vehicle at all, 0 - it won't move
+    /**
+     * Mass coefficient of the vehicle.
+     * Depends on wheel area.
```

Comment:
>  If the k_* family truly are implementation details they need to be encapsulated within vehicle and an alternative set of values provided as part of the public API.

Those exist: `acceleration`, `safe_speed` etc.

The specific results of `k_*` functions are implementation details, the functions themselves aren't. They have to be partially magical because they're imprecise approximations of real life data, tuned to return balanced values rather than to represent reality.

Not sure how do you see the structure of those. It's impossible to expose all the details of functions like `k_traction` (and even if it was possible, it would be a giant block of text that shouldn't be in the documentation anyway). while for `k_mass` and `k_friction` it would be just a block of implementation details.

As for documenting where are those used: this doesn't happen pretty much anywhere else. Check out `item.h`: there are nearly no comments like "it is used in x". Most are just "it returns x", "it changes the item so that it is x", "it returns the x property of the item".